modules/nf-core/gsea/gsea/main.nf

```diff
@@ -0,0 +1,80 @@
+    tuple val(meta), path("*/[!gene_set_size|gsea_report|ranked_gene_list]*.tsv")                                                           , emit: gene_set_tsv, optional: true
+    tuple val(meta), path("*/[!gsea_report|heat_map_corr_plot|index|pos_snapshot|neg_snapshot]*.html")                                      , emit: gene_set_html, optional: true
+    tuple val(meta), path("*/[!butterfly|enplot|global_es_histogram|gset_rnd_es_dist|heat_map|pvalues_vs_nes_plot|ranked_list_corr]*.png")  , emit: gene_set_heatmap, optional: true
+    tuple val(meta), path("*/*_snapshot*.html")                                                                                             , emit: snapshot, optional: true
+    tuple val(meta), path("*/enplot*.png")                                                                                                  , emit: gene_set_enplot, optional: true
+    tuple val(meta), path("*/gset_rnd_es_dist*.png")                                                                                        , emit: gene_set_dist, optional: true
+    tuple val(meta), path("*/*.zip")                                                                                                        , emit: archive, optional: true
+    path "versions.yml"                                                                                                                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def VERSION = '4.3.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    # Run GSEA
+    gsea-cli GSEA \
+        -res $gct \
+        -cls ${cls}#${target}_versus_${reference} \
+        -gmx $gene_sets \
+        -out . \
+        --rpt_label $prefix \
+        $args
+
+    # Move things out of the timestamped folder
+    mv ${prefix}.Gsea.* ${prefix}
+
+    # Other files have the annoying timestamp prefix too, but may be referred
+    # to from other files (e.g. the index.html). Symlinked here to provide
+    # consistent paths for the nf-core CI
+
+    pushd $prefix
+    ln -s ${prefix}.Gsea.*.rpt ${prefix}.Gsea.rpt
+    (ls *.zip 2>/dev/null || true) | while read -r l; do
+        ln -s \$l ${prefix}.Gsea.rpt.zip
+    done
+
+    ln -s \$(ls ranked_gene_list_${target}_versus_${reference}_*.tsv) ranked_gene_list_${target}_versus_${reference}.tsv
+    ln -s \$(ls gsea_report_for_${reference}_*.html) gsea_report_for_reference_${reference}.html
+    ln -s \$(ls gsea_report_for_${reference}_*.tsv) gsea_report_for_reference_${reference}.tsv
+    ln -s \$(ls gsea_report_for_${target}_*.html) gsea_report_for_target_${target}.html
+    ln -s \$(ls gsea_report_for_${target}_*.tsv) gsea_report_for_target_${target}.tsv
+    popd
```

Comment:
Thanks for this. The looping had been made more difficult by an earlier workaround to label things as 'reference' based on `args`, for the output pattern to pick up. That's no longer necessary since I can refer directly to `reference` and `target` since they're channel inputs- hooray! Obviously I shouldn't have been using subshells regardless.

Now fixed, and looped.